android/src/main/kotlin/com/mparticle/mparticle_flutter_sdk/MparticleFlutterSdkPlugin.kt

```diff
@@ -135,6 +138,69 @@ class MparticleFlutterSdkPlugin: FlutterPlugin, MethodCallHandler {
           jsonObj.put(entry.key.toString(), value)
         }.let { result.success((it ?: JSONObject()).toString()) }
       "logCommerceEvent" -> this.logCommerceEvent(call, result)
+      "getGDPRConsentState" -> getUser(call, result)?.let { user ->
+        user.consentState.gdprConsentState
```

Comment:
I'm not super familiar with kotlin syntax - should there be some `?` on user.consentState.gdprConsentState to ensure each exist?

Reply:
good call out, looking back on this it looks kinda sus to me as well....turns out its all good though. 

`getUser()` returns a nullable value, but we account for that with the `?.let` (so `user` is a nonnull `MParticleUser). From there, `consentState` is also nonnull, as well as `gdprConsentState`...that one is a little weird, ccpaConsentState is nullable since it is a single instance (either its there or it's not) but gdpr is a Map/Collection, so if none have been set then its going to return an empty map rather than `null`